src/libostree/ostree-mutable-tree.c

```diff
@@ -332,6 +332,51 @@ ostree_mutable_tree_replace_file (OstreeMutableTree *self,
   return ret;
 }
 
+/**
+ * ostree_mutable_tree_remove:
+ * @self: Tree
+ * @name: Name of file or subdirectory to remove
+ * @allow_noent: If @FALSE, an error will be thrown if @name does not exist in the tree
+ * @error: a #GError
+ *
+ * Remove the file or subdirectory named @name from the mutable tree @self.
```

Comment:
Nitpick: This should have a `Since: 2018.9` line (and upstream should have requested that). More important to get that in upstream rather than here. I’ll file a merge request upstream in a bit, once this has [actually merged upstream](https://github.com/ostreedev/ostree/pull/1724#issuecomment-423214368).